apps/wallet-mobile/src/components/Modal/ModalContext.tsx

```diff
@@ -36,8 +36,10 @@ export const ModalProvider = ({
   const navigation = useNavigation()
   const actions = React.useRef<ModalActions>({
     closeModal: () => {
-      dispatch({type: 'close'})
-      navigation.goBack()
+      if (navigation.getState().routes.slice(-1)[0].name === 'modal') {
```

Comment:
2 points here:
```const lastRouteName = navigation.getState().routes.slice(-1)[0].name``` for readability, and if I'm not mistaken react-navigation eventually adds to the name `key-<random>-<name>` but I don't recall it precisely, if it still does, then I'd say to check with something like "endsWith", print the navigation.getState(), and give it a check please, maybe I'm just delusional.

Reply:
key-<random>-<name> is the key, the name is just the name